docs/paradigm-layouts.md

```diff
@@ -0,0 +1,223 @@
+line that only consists of whitespace (including the tab character).
+This is equivalent to a row that consists of `C` **empty** cells (see
+below). A blank row **SHOULD** consist only of zero or more tab
+separators.
+
+A row can either be a:
+
+ - **header row**: its first cell must be a **header label**, followed
+   by whitespace.
+ - **content row**: a row with any other kind of cells
+
+A **label** is a single cell with one or more _tags_. Each tag
+in the label begins with a **prefix**, a **space character**, and then
+the tag proper. Each tag within a label is separated by a single space.
+
+These are the different prefixes:
+
+ - `#`: header labels: label that describes the entire pane it appears in
+ - `_`: row labels: describes the row in the current pane
+ - `|`: column labels: describes the column in the current pane
+
+For example, `# Past # Indicative` is a **header** label that has two
+**tags**: `Past` and `Indicative`. The **relabelling** system is
+intended to interpret the tags to display a user-facing string.
+
+A **cell** can be a **label** (as described above), an **empty** cell
+(consisting of zero or more whitespace characters, a **missing** form
+cell, consisting exactly of the string `--` or a **wordform** cell.
+A parser **MUST** attempt interpreting the cell as a **label cell**, an
+**empty cell**, or a **missing form** before it may interpret the cell
+as a **wordform cell**.
+
+A wordform cell is either a static wordform cell or a dynamic wordform
+cell. A _dynamic_ wordform cell contains exactly one `${lemma}`
+placeholder. The placeholder must be **filled** by an external system,
+and the cell may be substituted with a generated wordform. A _static_
+wordform cell does not contain a placeholder, and can be displayed to
+users verbatim, without any substitution.
+A paradigm layout is a _dynamic layout_ when it contains one or more
+_dynamic wordform cells_. A paradigm layout is a _static layout_ when
+it contains zero _dynamic_ wordform cells. A _dynamic layout_ **MUST**
+be _filled_ before the layout is ready for presentation.
+
+Aside from the presence or absence of the `${lemma}` placeholder, the
+exact syntax of a wordform cell is defined by the implementation.
+
+### Partial Grammar
+
+Here is a partial W3C EBNF grammar of the layout file specification.
+The syntax of `WordformCell` and `Tag` are implementation-dependent.
+
+```ebnf
+Layout ::= (Row NL)+
+
+Row ::= BlankRow | HeaderRow | ContentRow
+
+HeaderRow ::= HeaderLabel (TAB EmptyCell)*
+ContentRow ::= ContentCell (TAB ContentCell)*
+
+ContentCell ::= RowLabel
+              | ColumnLabel
+              | MissingForm
+              | EmptyCell
+              | WordformCell
+BlankRow ::= EmptyCell (TAB EmptyCell)*
+
+HeaderLabel ::= ('#' SP Tag)+
+RowLabel ::= ('_' SP Tag)+
+ColumnLabel ::= ('|' SP Tag)+
+MissingForm ::= '-' '-'
+EmptyCell ::= SP*
+
+TAB ::= #x09
+NL  ::= #x0A
+SP  ::= #x20
+```
+
+### Example
```

Comment:
This is super useful! Might even be worth putting this towards the top, followed by the explainer.